akka-http-testkit/src/main/scala/akka/http/scaladsl/testkit/RouteTestResultComponent.scala

```diff
@@ -95,7 +97,11 @@ trait RouteTestResultComponent {
     private def failNeitherCompletedNorRejected(): Nothing =
       failTest("Request was neither completed nor rejected within " + timeout)
 
-    private def awaitAllElements[T](data: Source[T, _]): immutable.Seq[T] =
-      data.grouped(100000).runWith(Sink.head).awaitResult(timeout)
+    private def awaitAllElements[T](data: Source[T, _]): immutable.Seq[T] = {
+      data.grouped(100000).runWith(Sink.head).recover({
+        case e: NoSuchElementException ⇒ Nil
+      })(ExecutionContexts.sameThreadExecutionContext)
+        .awaitResult(timeout)
+    }
```

Comment:
I was just going to reply that the current behavior is consistent with the stdlib.

~~~scala
scala> Nil.head
java.util.NoSuchElementException: head of empty list
  at scala.collection.immutable.Nil$.head(List.scala:420)
  ... 33 elided
~~~